src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasRelationSlicer.java

```diff
@@ -287,68 +293,81 @@ private boolean containsSlicedMember(final Iterable<RelationMember> members)
+            // if this outer's last location is closer to the original first coordinate of the
+            // linestring, then replace it
+            if (!firstSnapped || outerLineLast.distanceTo(originalFirst)
+                    .isLessThan(newLineLocations.get(0).distanceTo(originalFirst)))
             {
-                // Add the scaled location to the line
-                final Location scaledLocation = JTS_LOCATION_CONVERTER.backwardConvert(
-                        getCoordinateToPointMapping().getScaledCoordinate(pointCoordinate));
-                newLineLocations.add(scaledLocation);
-
-                // If the location doesn't have a point in the original Atlas or in the cache,
-                // make a new Point and add it both
-                if (Iterables.isEmpty(getStartingAtlas().pointsAt(scaledLocation))
-                        && !getCoordinateToPointMapping().containsCoordinate(pointCoordinate))
-                {
-                    final Map<String, String> newPointTags = createPointTags(pointLocation, false);
-                    newPointTags.put(SyntheticBoundaryNodeTag.KEY,
-                            SyntheticBoundaryNodeTag.YES.toString());
-                    final long newPointIdentifier = createNewPointIdentifier(
-                            pointIdentifierGenerator);
-                    final CompletePoint newPoint = new CompletePoint(newPointIdentifier,
-                            scaledLocation, newPointTags, new HashSet<Long>());
-                    // Store coordinate to avoid creating duplicate points
-                    getCoordinateToPointMapping().storeMapping(pointCoordinate, newPointIdentifier);
-                    // Save the point to add to the rebuilt atlas
-                    this.changes.add(FeatureChange.add(newPoint, getStartingAtlas()));
-                }
+                newLineLocations.set(0, outerLineLast);
+                firstSnapped = true;
             }
-            else
+            if (!lastSnapped || outerLineFirst.distanceTo(originalLast).isLessThan(
+                    newLineLocations.get(newLineLocations.size() - 1).distanceTo(originalLast)))
             {
-                newLineLocations.add(pointLocation);
+                newLineLocations.set(newLineLocations.size() - 1, outerLineFirst);
+                lastSnapped = true;
             }
         }
 
+        if (newLineLocations.get(0).distanceTo(originalFirst).isGreaterThan(SNAP_DISTANCE))
+        {
+            logger.error(
+                    "For relation {}, snapped first location by {} feet, not adding synthetic line",
+                    relationIdentifier, newLineLocations.get(0).distanceTo(originalFirst).asFeet());
+            return null;
+        }
+
+        if (newLineLocations.get(newLineLocations.size() - 1).distanceTo(originalLast)
+                .isGreaterThan(SNAP_DISTANCE))
+        {
+            logger.warn(
```

Comment:
Any reason this one is a `warn` when the other one is an `error`?

Reply:
nope, oversight!